### PR TITLE
Player rocket emitter

### PR DESCRIPTION
**Pull request by Miho42**
After loads of hardship, and incredibly great amounts of tears and also some sweat,

I present to you:

A rocket with a fire-thingy made with an emitter....

And it works.

### CODE REVIEW

File: my_game.py [hunk 1]
```diff
@@ -207,6 +207,7 @@ def __init__(self, width, height):
         self.player_shot_list = None
         self.asteroids_list = None
         self.UFO_list = None
+        self.emitters_list = None
```
Comment (tobiasbp) [inline my_game.py:210]: Don't think we need this?

File: my_game.py [hunk 2]
```diff
@@ -266,13 +267,18 @@ def setup(self):
...
 
-
         # Create a Player object
         self.player_sprite = Player(
             center_x=PLAYER_START_X,
             center_y=PLAYER_START_Y
         )
 
+        # Player rocket emitter in variable
+        self.player_rocket = None
```
Comment (tobiasbp) [inline my_game.py:277]: Can this be _player_rocket_emitter_ for clarity?

File: my_game.py [hunk 3]
```diff
@@ -306,9 +316,21 @@ def on_draw(self):
             "LIVES: {}".format(self.player_lives ),  # text to show
             10,                  # X position
             SCREEN_HEIGHT - 50,  # Y position
-            arcade.color.WHITE    # color of text
+            arcade.color.WHITE   # color of text
         )
 
+    def get_player_rocket(self, player):
```
Comment (tobiasbp) [inline my_game.py:322]: Let us know what this function does like this:

```
"""
Some ducumentation
"""
```

File: my_game.py [hunk 4]
```diff
@@ -374,7 +397,20 @@ def on_update(self, delta_time):
         # Player rocket engine
         if self.up_pressed:
             self.player_sprite.player_thrust()
-        
+
+            # Only make rocket_controller if one doesn't already exist
+            if self.player_rocket is None:
```
Comment (tobiasbp) [inline my_game.py:402]: Maybe we could always have the _self.player_rocket_, but have it with a controller which emits 0 particles? Then we can set a new controller when the thrust key is ressed?

File: my_game.py [hunk 5]
```diff
@@ -282,13 +282,18 @@ def setup(self):
...
         self.player_sprite = Player(
             center_x=PLAYER_START_X,
             center_y=PLAYER_START_Y
         )
 
+        # Player rocket emitter in variable
+        self.player_rocket_emitter = None
+
+        # Player rocket emitter_controller in variable
+        self.player_rocket_controller = arcade.EmitterIntervalWithCount(0, 0)
```
Comment (tobiasbp) [inline my_game.py:295]: Let us know why you create this weird looking controller with 0 values?

File: my_game.py [hunk 6]
```diff
@@ -300,6 +305,10 @@ def on_draw(self):
         # Draw the player shot
         self.player_shot_list.draw()
 
+        if self.player_rocket_emitter is not None:
+        # Draw player rocket
```
Comment (tobiasbp) [inline my_game.py:309]: Incorrect comment. Weird indentation.

**Review by tobiasbp — CHANGES_REQUESTED** [review]
Minor changes needed.

File: my_game.py (hunk 4 above)
Comment (Miho42) [reply my_game.py]: player_rocket has been renamed "player_rocket_emitter"

The emitter is not the one controlling particle stuff, so I tried propping the emitter code into the variable, but the game froze, so I assumed it wasn't quite the right thing to do, Elsa got angry every time (ha ha very fun pun, the game _froze_)

**Review by tobiasbp — APPROVED** [review]

File: my_game.py [hunk 7]
```diff
@@ -239,6 +241,57 @@ def update(self):
...
+    It is possible to start and stop this emitter
+    """
+    def __init__(self,
+            target: arcade.Sprite,
+            particle_lifetime: float = 0.5,
+            noise: int = 15,
+            offset: Tuple[int] = (0, 6),
+            emit_interval: float = 0.01,
+            particle_count: int = 30,
+            start_alpa: int = 100):
```
Comment (Ashthecool) [inline my_game.py:255]: Ikke godt nok. Hvad i alverden er denne fejl. Hvem kan skrive alpa istedet for alpha. Altså helt seriøst.

**Pull request merged**